### PR TITLE
[Core] Make core_commands.py comply with #4048

### LINKED ISSUE #4048: [Tracking] An initiative to make codebase racially neutral
Making this issue to take place instead of #4047.

There is a few things to ensure while taking these steps.

- This should not be done as one bulk PR
  - Each cog that has this applied to it should have individual PRs
  - An Individual PR for core changes should exist
- We shouldn't make any breaking changes/changes that would confuse the existing user base
- Before (if at all) removing all mentioned of `blacklist` and `whitelist` we should give users some grace period, because if not this would be a breaking change, that's not compliant with our guarantees.
- When changing internals we should ensure everything they change is also changed.
- When changing config defaults/existing values, we should ensure a data migration is performed.

Currently the following uses either `whitelist` or `blacklist`:
- Audio Cog (redbot/cogs/audio/core/__init__.py)
  - Audioset commands (redbot/cogs/audio/core/commands/audioset.py)
  - Audio Utils (redbot/cogs/audio/core/utilities/validation.py)
  - Audio Lavalink Manager (redbot/cogs/audio/manager.py)
- Bot Core (redbot/core/bot.py)
- Bot Core commands (redbot/core/core_commands.py)
- Bot Global checks (redbot/core/global_checks.py)
- Bot Setting cache (redbot/core/settings_caches.py)
- Permissions converters (redbot/cogs/permissions/converters.py)

*Docs should auto update once the above are update.*

The permission cogs currently allows "blacklist" and "whitelist" as a user input for it's converters I don't think this should ever be changed, although we should add the new keywords to this converter to also accept them.

---

We should focus on changing front facing commands first:
- [ ] Audioset commands (redbot/cogs/audio/core/commands/audioset.py)
- [x] Bot Core commands (redbot/core/core_commands.py)
- [ ] Permissions converters (redbot/cogs/permissions/converters.py)

*Note: We should rename the user facing commands only at this stage, while aliasing the old names*

---

Once the above are worked on we should then look at potentially implementing this in the backend and config.
- [ ] Audio Cog (redbot/cogs/audio/core/__init__.py)
  - [ ] Audioset commands (redbot/cogs/audio/core/commands/audioset.py)
  - [ ] Audio Utils (redbot/cogs/audio/core/utilities/validation.py)
  - [ ] Audio Lavalink Manager (redbot/cogs/audio/manager.py)
- [ ] Bot Core (redbot/core/bot.py)
- [x] Bot Core commands (redbot/core/core_commands.py)
- [ ] Bot Global checks (redbot/core/global_checks.py)
- [ ] Bot Setting cache (redbot/core/settings_caches.py)

---
Note: Due to audio being in a major rewrite currently I would ask that if you choose to pick up this PR to work on it, don't touch audio. as I will be following the above in Audio anyways as part of the rewrite.

Also if I missed anything on the summary above please feel free to flag it.

### PR DESCRIPTION
**Pull request by Dav-Git**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
This PR starts work on #4048 but only takes care of ``redbot/core/core_commands.py`` renaming all instances of "blacklist" to "denylist" and "whitelist" to "allowlist".

### CODE REVIEW

**Comment by Drapersniper** [comment]
Thanks!
For the sake of conflicts and to not risk delaying #4045 any further. Going to mark this PR as blocked by that one,  (since this bad boy will cause a lot of other PRs to become conflicting once it is merged)

**Comment by Dav-Git** [comment]
Hey look at that, no conflicts! Or do I need to do some magic now the DATA API is merged?

**Comment by Drapersniper** [comment]
Can I be a complete and utter ass ...  at least for myself `blocklist` makes more sense to me than `denylist`

So can you change it please ... (Before you do give it .. 24 hours and see if anyone speaks up agaisnt/in favour  though)

**Comment by Dav-Git** [comment]
Sorry Draper, I totally didn't forget about this at all. Got it switched out (properly TM) but added ``denylist`` as an alias because I like the word. Does that work for you?

**Comment by Drapersniper** [comment]
In the surface I'm fine with it staying as an alias ... however I do have concerns about it clogging the namespace for commands (that's why we avoid aliases to bare commands and tend to apply them only to sub commands that way a single core command isn't reserving multiple commands in the namespace).

If no one is Has an issue with it I'm fine with it in this case.

**Comment by Dav-Git** [comment]
The thought process here was that anything implementing similar allow/blocklist functionality is already limited by the unavailability of the allowlist command as well as the danger of being confused. Most 3rd party cogs would solve this by putting such commands behind a settings command group. I think while I generally agree that the command namespace should be kept as clear as possible this does not take away a large chunk of the practically usable spectrum.

**Review by Drapersniper — DISMISSED** [review]
LGTM...

I will say though, im not a fan of the "Local blocklist management commands.' usage, specifically `Local ` here as without context I'm not sure what local means. IMO it should be `server`  however in other places in the bot we do use local so won't block or nitpick on this PR because of it.

**Comment by Dav-Git** [comment]
Do NOT merge this, just found a bunch of instances of denylist my search and replace didn't catch and it's inconsistent in some places.

To adress latest review comment, I chose local in the strings where I did due to the function names. Let me know if you want that changed.

File: redbot/core/core_commands.py
Comment (Drapersniper) [inline redbot/core/core_commands.py]: Did you mean to do this here and in line 1994?

Comment (Dav-Git) [reply redbot/core/core_commands.py]: Close but not quite. One sec

File: redbot/core/core_commands.py
Comment (Drapersniper) [inline redbot/core/core_commands.py]:
```suggestion
            await ctx.send("Blocklist is empty.")
```

**Review by Drapersniper — DISMISSED** [review]
Only change needed here

**Review by Drapersniper — APPROVED** [review]
LGTM, Thanks Dav.

**Pull request merged**
